CARDS-1430: Regression: Vocabulary term displayed value is its id instead of the label

veronikaslc opened the pull request:
The problem appeared to be with the priorities order of the `ResourceJsonProcessors`. `AnswerOptionsLabelProcessor` was always running after `VocabularyLabelProcessor` overwriting the right processed json label value with the term id instead of the value. 

However there is a room for the same regression in future. `AnswerOptionsLabelProcessor` can run on any answer node with options, its conditions are not unique. Every processor that extends it has to be explicitly excluded from it's `leave()` function condition like in this PR, by filtering out corresponding node, to avoid double-running. Even explicit numerical priorities setting is not the optimal solution cause many processors can still run for the same answer overwridig results of each other.

Commit:
CARDS-1430: Regression: Vocabulary term displayed value is its id instead of the label

sashaandjic reviewed (approved):
tested
![image](https://user-images.githubusercontent.com/23243244/138791035-56aeeee1-34cd-4f89-8230-7fe4f85e664d.png)
![image](https://user-images.githubusercontent.com/23243244/138791067-2a099b9b-727d-4ee5-a2d5-29489e3755bb.png)

sdumitriu committed:
[cleanup] Improved comments and codestyle

sdumitriu committed:
CARDS-1430: Regression: Vocabulary term displayed value is its id instead of the label
Rollback unneeded changes

sdumitriu committed:
CARDS-1430: Regression: Vocabulary term displayed value is its id instead of the label
Better fix: make sure the vocabulary labels are added after the answer option labels

sdumitriu commented:
> I'm afraid it won't prevent it from running.

It doesn't have to prevent it from running, it is only needed for `VocabularyLabelProcessor` to run after `AnswerOptionsLabelProcessor`. First `AnswerOptionsLabelProcessor` sets a `displayedValue` that contains predefined option labels and raw vocabulary paths, then `VocabularyLabelProcessor` sets a new `displayedValue`, discarding the one set before, that contains both predefined option labels and vocabulary labels. The only disadvantage is that `AnswerOptionsLabelProcessor` does its work for nothing, but it's just a minor waste of computation for the worthy price of better code isolation.

sashaandjic reviewed (approved):
tested
